kevm-pyk/src/kontrol/foundry.py

```diff
@@ -771,18 +775,71 @@ def _split_test(test: tuple[str, str | None]) -> tuple[str, str, str | None]:
             contract, method = test_name.split('.')
             return contract, method, id
 
-        init_problems = [_split_test(test) for test in tests]
+        llvm_definition_dir = foundry.llvm_library if use_booster else None
 
-        _apr_proofs: list[tuple[bool, list[str] | None]]
-        if workers > 1:
-            with ProcessPool(ncpus=workers) as process_pool:
-                _apr_proofs = process_pool.map(_init_and_run_proof, init_problems)
-        else:
-            _apr_proofs = []
-            for init_problem in init_problems:
-                _apr_proofs.append(_init_and_run_proof(init_problem))
 
-        apr_proofs = dict(zip(tests, _apr_proofs, strict=True))
+        def create_server() -> KoreServer:
+            return kore_server(
+                definition_dir=foundry.kevm.definition_dir,
+                llvm_definition_dir=llvm_definition_dir,
+                module_name=foundry.kevm.main_module,
+                port=port,
+                command=kore_rpc_command,
+                bug_report=br,
+                smt_timeout=smt_timeout,
+                smt_retry_limit=smt_retry_limit,
+            )
+
+        results = {}
+        futures: dict[str, Future] = {}
+        with KoreServerPool(create_server=create_server, max_workers=workers) as pool:
```

Comment:
`KoreServerPool` starts a server per worker, Unfortunately, servers are not stateless due to calls to `add-module`.

There are two solutions:

1. Do not use `KoreServerPool`, but have a separate server for each test. As the server can handle concurrent requests, having more than one active workers for a single test shouldn't cause an issue.
2. Add an endpoint `reset-module` and call it every time a worker picks up a new task. Then add back the modules necessary for the given test.

Reply:
There is now a system where it starts one server for each original/base test, and passes in the port for the appropriate server when starting subproofs. However, I am now realizing that we still want the servers to be started inside the worker threads, since startup is non-trivial.